### PR TITLE
Address review comments to #1650

### PR DESCRIPTION
**Pull request by ahoppen**
This addresses the review comments I made on #1650.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift [hunk 1]
```diff
@@ -431,7 +457,7 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
       return .skipChildren
     }
 
-    if node.leftSquareBracket.presence == .missing && node.rightSquareBracket.presence == .present {
+    if node.leftSquareBracket.isMissing && node.rightSquareBracket.presence == .present {
```
Comment (bnbarham) [inline Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift:460]: Personally I'm not a huge fan of introducing an accessor here. If you *really* want it then I'd also want a `isPresent`. Or if we've decided there's only ever going to be two values here, then changing the `.presence == .present` to `!isMissing` would be another option.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Tests/SwiftParserTest/RegexLiteralTests.swift [hunk 2]
```diff
@@ -751,13 +751,13 @@ final class RegexLiteralTests: XCTestCase {
...
   func testBinOpDisambiguation26() {
-    // FIXME: The diagnostic should be one character back
     assertParse(
       """
-      # 1️⃣/^ x/
+      #1️⃣ 2️⃣/^ x/
       """,
       diagnostics: [
-        DiagnosticSpec(message: "expected identifier in macro expansion", fixIts: ["insert identifier"])
+        DiagnosticSpec(locationMarker: "1️⃣", message: "extraneous whitespace after '#' is not permitted", fixIts: ["remove whitespace"]),
```
Comment (bnbarham) [inline Tests/SwiftParserTest/RegexLiteralTests.swift:759]: Bit weird to have both (1) and (2). Ideally we'd just have (2) with the trivia transferred to the trailing of the inserted identifier.

**Review by bnbarham — APPROVED** [review]

File: Tests/SwiftParserTest/RegexLiteralTests.swift (hunk 2 above)
Comment (ahoppen) [reply Tests/SwiftParserTest/RegexLiteralTests.swift]: Changed it in a separate commit to this PR.

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please clean test Windows

**Pull request merged**